src/extension.c

```diff
@@ -92,41 +105,14 @@ extension_event_trigger(PG_FUNCTION_ARGS)
 	if (!CALLED_AS_EVENT_TRIGGER(fcinfo))
 		elog(ERROR, "not fired by event trigger manager");
 
+	/* TODO: does this do anything useful? */
 	if (strcmp(trigdata->event, "ddl_command_end") == 0 &&
 		strcmp(trigdata->tag, "CREATE EXTENSION") == 0 &&
 		IsA(trigdata->parsetree, CreateExtensionStmt) &&
 		strcmp(((CreateExtensionStmt *) trigdata->parsetree)->extname, EXTENSION_NAME) == 0)
 	{
 		extstate = EXTENSION_STATE_CREATING;
 	}
-	else if (strcmp(trigdata->event, "ddl_command_start") == 0 &&
-			 strcmp(trigdata->tag, "DROP EXTENSION") == 0 &&
-			 IsA(trigdata->parsetree, DropStmt))
-	{
-		DropStmt   *stmt = (DropStmt *) trigdata->parsetree;
-		const char *extname = strVal(linitial(linitial(stmt->objects)));
-
-		if (strcmp(extname, EXTENSION_NAME) == 0)
-		{
-			extstate = EXTENSION_STATE_DROPPING;
-
-			/*
-			 * Save the transaction ID, so that we can avoid going from
-			 * INITIAL to CREATED in the transaction that issued DROP
-			 * EXTENSION.
-			 */
-			drop_transaction_id = GetCurrentTransactionId();
-
-			/*
-			 * Notify other backends that the extension was dropped. We do
-			 * this via the relcache invalidation mechanism in Postgres by
-			 * issuing an invalidation on a proxy table. Other backends will
-			 * see an invalidation event on the proxy table and then knows
-			 * they need to move back to INITIAL state.
-			 */
-			CacheInvalidateRelcacheByRelid(extension_proxy_oid);
```

Comment:
So it turns out that postgres adds an relcache invalidation message for the relid of any table it is dropping. The stacktrace is : 
```
* thread #1: tid = 0xd9b1e7, 0x000000010ae91eae postgres`RegisterRelcacheInvalidation(dbId=54515356, relId=54515644) + 14 at inval.c:498, queue = 'com.apple.main-thread', stop reason = breakpoint 2.1
  * frame #0: 0x000000010ae91eae postgres`RegisterRelcacheInvalidation(dbId=54515356, relId=54515644) + 14 at inval.c:498
    frame #1: 0x000000010ae91d51 postgres`CacheInvalidateHeapTuple(relation=0x000000010b372bf0, tuple=0x00007fff55346ac0, newtuple=0x0000000000000000) + 529 at inval.c:1174
    frame #2: 0x000000010a91efb9 postgres`heap_delete(relation=0x000000010b372bf0, tid=0x00007f8d128d9934, cid=106, crosscheck=0x0000000000000000, wait='\x01', hufd=0x00007fff55346b68) + 4073 at heapam.c:3345
    frame #3: 0x000000010a9201d6 postgres`simple_heap_delete(relation=0x000000010b372bf0, tid=0x00007f8d128d9934) + 70 at heapam.c:3378
    frame #4: 0x000000010a9c5e76 postgres`DeleteAttributeTuples(relid=54515755) + 182 at heap.c:1482
    frame #5: 0x000000010a9c6a0d postgres`heap_drop_with_catalog(relid=54515755) + 381 at heap.c:1848
    frame #6: 0x000000010a9c2d82 postgres`doDeletion(object=0x00007f8d14833b38, flags=0) + 242 at dependency.c:1155
    frame #7: 0x000000010a9c0773 postgres`deleteOneObject(object=0x00007f8d14833b38, depRel=0x00007fff55346e58, flags=0) + 163 at dependency.c:1057
    frame #8: 0x000000010a9c040b postgres`deleteObjectsInList(targetObjects=0x00007f8d1480d770, depRel=0x00007fff55346e58, flags=0) + 315 at dependency.c:246
    frame #9: 0x000000010a9c0575 postgres`performMultipleDeletions(objects=0x00007f8d1480d240, behavior=DROP_CASCADE, flags=0) + 261 at dependency.c:385
    frame #10: 0x000000010aa9daad postgres`RemoveObjects(stmt=0x00007f8d1405bf80) + 909 at dropcmds.c:134
    frame #11: 0x000000010ad11aad postgres`ExecDropStmt(stmt=0x00007f8d1405bf80, isTopLevel='\x01') + 205 at utility.c:1607
    frame #12: 0x000000010ad11243 postgres`ProcessUtilitySlow(parsetree=0x00007f8d1405bf80, queryString="drop schema public cascade;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, dest=0x00007f8d1405c328, completionTag="") + 7107 at utility.c:1486
    frame #13: 0x000000010ad0eeba postgres`standard_ProcessUtility(parsetree=0x00007f8d1405bf80, queryString="drop schema public cascade;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, dest=0x00007f8d1405c328, completionTag="") + 3114 at utility.c:819
    frame #14: 0x000000010b5a761e timescaledb.so`prev_ProcessUtility(parsetree=0x00007f8d1405bf80, queryString="drop schema public cascade;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, dest=0x00007f8d1405c328, completionTag="") + 110 at process_utility.c:34
    frame #15: 0x000000010b5a7580 timescaledb.so`timescaledb_ProcessUtility(parsetree=0x00007f8d1405bf80, queryString="drop schema public cascade;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, dest=0x00007f8d1405c328, completionTag="") + 496 at process_utility.c:90
    frame #16: 0x000000010ad0e25c postgres`ProcessUtility(parsetree=0x00007f8d1405bf80, queryString="drop schema public cascade;", context=PROCESS_UTILITY_TOPLEVEL, params=0x0000000000000000, dest=0x00007f8d1405c328, completionTag="") + 124 at utility.c:332
    frame #17: 0x000000010ad0dbf3 postgres`PortalRunUtility(portal=0x00007f8d1280c040, utilityStmt=0x00007f8d1405bf80, isTopLevel='\x01', setHoldSnapshot='\0', dest=0x00007f8d1405c328, completionTag="") + 371 at pquery.c:1193
    frame #18: 0x000000010ad0ce2f postgres`PortalRunMulti(portal=0x00007f8d1280c040, isTopLevel='\x01', setHoldSnapshot='\0', dest=0x00007f8d1405c328, altdest=0x00007f8d1405c328, completionTag="") + 527 at pquery.c:1342
    frame #19: 0x000000010ad0c46e postgres`PortalRun(portal=0x00007f8d1280c040, count=9223372036854775807, isTopLevel='\x01', dest=0x00007f8d1405c328, altdest=0x00007f8d1405c328, completionTag="") + 910 at pquery.c:815
    frame #20: 0x000000010ad07d4a postgres`exec_simple_query(query_string="drop schema public cascade;") + 1242 at postgres.c:1094
    frame #21: 0x000000010ad06f50 postgres`PostgresMain(argc=1, argv=0x00007f8d140045f0, dbname="single", username="postgres") + 2368 at postgres.c:4076
    frame #22: 0x000000010ac62b12 postgres`BackendRun(port=0x00007f8d138006c0) + 674 at postmaster.c:4285
    frame #23: 0x000000010ac61dcd postgres`BackendStartup(port=0x00007f8d138006c0) + 365 at postmaster.c:3959
    frame #24: 0x000000010ac60e25 postgres`ServerLoop + 597 at postmaster.c:1715
    frame #25: 0x000000010ac5e636 postgres`PostmasterMain(argc=15, argv=0x00007f8d11403800) + 5414 at postmaster.c:1323
    frame #26: 0x000000010ab8f11f postgres`main(argc=15, argv=0x00007f8d11403800) + 751 at main.c:228
    frame #27: 0x00007fff943af5ad libdyld.dylib`start + 1
    frame #28: 0x00007fff943af5ad libdyld.dylib`start + 1
```
The sql was: 
```
single=# select '_timescaledb_cache.cache_inval_extension'::regclass::oid;
   oid
----------
 54515644
(1 row)

Time: 0.534 ms
single=# drop schema public cascade;
```

In `CacheInvalidateHeapTuple` you can actually see that any modification to a pg_class tuple causes an invalidation event on the OID of the tuple (the relid of the corresponding table).